### ISSUE TITLE
[South-folderScanner] Impossible to scan a repository

### ISSUE THREAD

**Issue by victor-tndl** [OptimistikSAS/OIBus#992]
**Describe the bug**
I can't read a folder whatever the folder name and path and I have this error:
`FolderScanner.class(72) - 2020-12-17T15:19:50.008Z-error: The input folder /path/to/repo/OIBus/tests/csv is not readable: Error: SQLITE_ERROR: no such column: modified`

But before this error, the folderScnaner scan well the file if we consider the following logs.
`FolderScanner.class(46) - 2020-12-17T15:19:50.002Z-silly: FolderScanner activated on scanMode: every10Seconds.
FolderScanner.class(85) - 2020-12-17T15:19:50.005Z-silly: checkConditions:csvFilename.csv match regexp
FolderScanner.class(89) - 2020-12-17T15:19:50.006Z-silly: checkConditions: mT:1608218316332.4749 + mA 1000 < ts:1608218390005  = true
FolderScanner.class(91) - 2020-12-17T15:19:50.006Z-silly: checkConditions: csvFilename.csv match age
`

Just to let you know I have this bug since I use the fork project.

**To Reproduce**
I just created a south folder scanner with path (csv).

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Browser: mozilla firefox
 - Version: 83.0

**Smartphone (please complete the following information):**

**Additional context**
I have tried to change the permission for the folder and the file but no improvements...

**Comment by jfhenon** [comment]
Can you provide the oibus.json? (remove sensitive informations before).

**Comment by victor-tndl** [comment]
```json
{
    "engine": {
        "engineName": "OIBus",
        "port": 2223,
        "user": "admin",
        "password": "n2CD1lHQhU4jjFRJ+X/RwfZvu3pFiTkHwgOPXIpRsthL2tAWicfOpTpVIVQeR+2ejr92zc/uV1YDy72JUcJ3sM8vqHXkPeJS8E48YcY2tdj6mDIoG8P132S1TO9VsfrPHCLWP3K7Q5gSZJrXOCn0BknFJ4efsaqobX9Ph70hG+33rVwkiO3G1pHNEYkViv1Zz1lum22LajUS2MYvHFortUEJFcO7n0tTd4mfIDvxpy0Uplw/rWW9HBDD2RMQyQfGPec7KgNr7vMWMhYO8wZ8twoyNZ5E6ZXzQb0R7NoTadjUwhoxyRWE2VP52AUI9Zym9lV+VInRnk2wJa3XrusUjZ/jrhP/sgI3iU0zpsI4fKO422mybz98JMtnRu9IJhYzHN2FK11SK+Z+MEVvvKX8AWX9B2QeHzE8jTuJ5//ASRcXRr4B1iQzQqJtJJkTDoUqXiiNHvglKPuIa1YaCkXVBEIYWPBbEiLEOJwzS6WPho/LkUAyVC0lg9T8qLUCsGXcQv4X2CtZ0OaIWn9S0/3G8oAnreoCGS8WOs6cc3GLPMoTGyggHydz/SJwbt4W2JP1cerjJtAnJYQlCvXkHxhi+v3BxHwk9SI1WiveLy1bcY1sVTlmnlZPHHaPip240Fz3syFwz7wiDEL5NfawuhbhCYWenvnmZgauVpW3a8frXps=",
        "filter": [
            "127.0.0.1",
            "::1",
            "::ffff:127.0.0.1"
        ],
        "safeMode": false,
        "logParameters": {
            "consoleLevel": "silly",
            "fileLevel": "silly",
            "filename": "./logs/journal.log",
            "maxsize": 1000000,
            "maxFiles": 5,
            "tailable": true,
            "sqliteLevel": "silly",
            "sqliteFilename": "./logs/journal.db",
            "sqliteMaxFileSize": 50000000
        },
        "caching": {
            "cacheFolder": "./cache",
            "archiveMode": "delete",
            "archiveFolder": "./cache/archived/"
        },
        "scanModes": [
            {
                "scanMode": "everySecond",
                "cronTime": "* * * * * *"
            },
            {
                "scanMode": "every10Seconds",
                "cronTime": "* * * * * /10"
            },
            {
                "scanMode": "every1Min",
                "cronTime": "* * * * *"
            },
            {
                "scanMode": "every10Minutes",
                "cronTime": "* * * * /10"
            },
            {
                "scanMode": "listen",
                "cronTime": "listen"
            }
        ],
        "proxies": [],
        "aliveSignal": {
            "enabled": true,
            "host": "http://localhost:1880",
            "endpoint": "/ping/oibus",
            "authentication": {
                "type": "Custom",
                "username": "test",
                "password": "AN/tN883nKoMPctehTN0ANO0u57ZO425cYBnLysYgKcldt9e3rerrOWzNnC+Dsg3EOp6AiP39KGbtBtCu1pyx7pgN2AVTCXgxBy0XycAtvlAE5Rydi+YDzSAP5QSaQt5mYYCNo/YXgxO/udPhpt/yNsSIRDq7AW5hFQSt98AKCrdE53w2pTmD7l5Nhi8v36/mCffMfMlmZXBUqWEwZqay8rQObf026APq+glYAwdnQn0zWmdBs9HXLjTc92vax99kgzjO9n9CjgZ3W303pNjezrt4tuMueq4EY+cXdordZ74Bf8bBYnKF7ACaVrIj9tTUjsffxLX3d+p4We6qNG0avBxgwhwZrFqqyMDGn48HcUYkrALNEjfgXiu7H4zrNBDAVYLYHKSdSzXAvlNGbF9u4fhtCgI6SgKkx1Q2mpFpeNbdeBtWBZpx3/O9yKYwOXPUvZfnOULVpxyY7Q+yPgeX2iYVuZSwO9A4iiw2mKuH3DN6gyiu/YJSQSHQzh3Ei3ltVdcT/RT+6dPbsM53gqzQ7pVVDl//2uNHVGeQilNbFnXwJNSC57hMTEG0Wasdco18piai3Z9QQaf/qdULPrLNpotapLyyXvV+8FbqjPE3jqhewX8DcQaf7t4PPdvdU4xHo0KDLNt1MFyaVK5qYmYc570TVJ22woqW8xVIRQGJQI=",
                "headerValue": "{{notEncrypted}}tete",
                "headerName": "test",
                "secret": ""
            },
            "id": "1",
            "frequency": 60,
            "proxy": "",
            "verbose": true
        },
        "httpRequest": {
            "stack": "fetch",
            "timeout": 30,
            "retryCount": 3
        }
    },
    "south": {
        "dataSources": [
            {
                "dataSourceId": "FolderScanner",
                "protocol": "FolderScanner",
                "enabled": true,
                "FolderScanner": {
                    "preserve": true,
                    "minAge": 1000,
                    "compression": false,
                    "inputFolder": "csv",
                    "regex": ".csv"
                },
                "points": [],
                "scanMode": "every10Seconds",
                "logParameters": {
                    "consoleLevel": "engine",
                    "fileLevel": "engine",
                    "sqliteLevel": "engine"
                }
            }
        ]
    },
    "north": {
        "applications": [
            {
                "applicationId": "CSVToHttp",
                "api": "CsvToHttp",
                "enabled": true,
                "CsvToHttp": {
                    "authentication": {
                        "headerValue": "{{notEncrypted}}token",
                        "headerName": "X-auth-token",
                        "type": "Custom",
                        "password": "",
                        "username": ""
                    },
                    "secretKey": "h8cFGOJ6CuhvH6Buz6ungoadbzqOKkIDgqc3WBcjomCtdbAkVU8rwT9Y0aog/1bnp8jfMW8ptURbcCP8IGviSHMadjReJmsYnH924pBh6v2+FDTkurjrUiEwCHRTe+by3+Jwr8AHWiCQQss4DgUpxL/6KIj9Q/YhqoUWuVE7t1GI0tWDkayU6CACM+fP/yEjfaBjZ4VMyca2U0MkPfCZkyTxvIDkFBL3ixMJ5nXEXkm69eKMwz1Lu9KTUfgKhpFCbKpFjNGSa5/FDX5k1b/s9C84qki4ZQLAEPBuVpelgOtuT+HUH+OcExz1O3BvA9ek2e3gBNkxJDg3iJ6AUfVj3chXkCtgwuCqTfOi9JG/NsWxTKlxeK0ZJwhDXjN2v57fsoT5M3gAB5RlWdOannNt1ihSpMeF6atXYWL4JVYsTJCs3AANhyFsyZwF24Jinqex8Xd6Tb9Sy9oZtV5C6gK+XdYZ+orzBJmbswYC6oEPqeJx8gsL+PLcegSGClXeS0KWEk4f0YfEdvrXMljsEF4yofCAwzykfYcp2KbRlklLTXomTlkS3kEYcfyNSrn8L361HdHK+UEXTW/iYz4OmukKvUkPQZnopUhrgOndtxe1OtgUXQwXZolBweOaA2uxsf+/8261TV56Dseq/TEHVl71nKBcSO5Np27Naf8WpQBnH+g=",
                    "applicativeHostUrl": "https://localhost:1880/ping/oibus",
                    "requestMethod": "POST",
                    "httpAuthentication": "Credentials",
                    "identification": "g",
                    "proxy": "",
                    "mapping": [
                        {
                            "csvField": "Identification",
                            "httpField": "id",
                            "type": "integer"
                        },
                        {
                            "csvField": "Description",
                            "httpField": "des",
                            "type": "string"
                        },
                        {
                            "csvField": "Date de création du ticket",
                            "httpField": "date",
                            "type": "date (ISO)"
                        }
                    ]
                },
                "caching": {
                    "sendInterval": 10000,
                    "retryInterval": 5000,
                    "groupCount": 1000,
                    "maxSendCount": 10000
                },
                "subscribedTo": []
            },
            {
                "applicationId": "Console",
                "api": "Console",
                "enabled": true,
                "Console": {
                    "verbose": true
                },
                "caching": {
                    "sendInterval": 10000,
                    "retryInterval": 5000,
                    "groupCount": 1000,
                    "maxSendCount": 10000
                },
                "subscribedTo": [],
                "logParameters": {
                    "consoleLevel": "silly",
                    "fileLevel": "silly"
                }
            }
        ]
    },
    "schemaVersion": 17
}
```

**Comment by jfhenon** [comment]
OK, I could reproduce your issue.

**Comment by jfhenon** [comment]
@kukukk When I moved the configdb code (used in OPCUA/DA/SQL south handlers) in the protocolHandler, this created an issue for FolderScanner as this one use a different table. This is why the column "modified" could not be found. 
If you stop using preserveFiles, this bug won't appear so this is a temporary workaround for @victor-tndl .

Please look at the branch fix-issue#992 that surcharge the standard connect (so we don't create the default configDB). I think this should fix the issue. However, another approach maybe to use the standard configDB also in FolderScanner?

**Comment by kukukk** [comment]
ConfigDB uses a key-value approach to store some data. Actually, this data is the lastCompletedAt information for OPCUA/DA/SQL south. The FolderScannerDB contains the filename and the modified time to keep track of the last time a given file was scanned in case when preserveFile is checked. So, the main structure is kinda the same and they could be merged.
